clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -382,6 +382,12 @@ private void setSaslState(SaslState saslState, AuthenticationException exception
         } else {
             this.saslState = saslState;
             LOG.debug("Set SASL server state to {} during {}", saslState, reauthInfo.authenticationOrReauthenticationText());
+            if (saslState == SaslState.COMPLETE) {
+                // make sure that any pending write is actually sent.
+                if (reauthInfo.reauthenticating()) {
+                    transportLayer.addInterestOps(SelectionKey.OP_WRITE);
```

Comment:
If there is nothing to send, we will need to reset this somewhere (or set it here only if it was set before reauth). Otherwise poll may return early until this write interest is reset, even though there is nothing to read or write for this channel. Can we check that this doesn't cause unnecessary return from poll when there is nothing to send from the broker?